pkg/catalogv2/git/git.go

```diff
@@ -269,54 +269,79 @@ func (r *Repository) getCurrentCommit() (plumbing.Hash, error) {
 	return headRef.Hash(), nil
 }
 
+// checkoutCommit will checkout to a given commit and leave the local repo at a Detached HEAD state.
+func (r *Repository) checkoutCommit(commit string) error {
+	checkOpts := gogit.CheckoutOptions{
+		Hash: plumbing.NewHash(commit),
+	}
+
+	wt, err := r.repoGogit.Worktree()
+	if err != nil {
+		return fmt.Errorf("checkout failure to open worktree: %w", err)
+	}
+
+	err = wt.Checkout(&checkOpts)
+	if err != nil {
+		return fmt.Errorf("checkout failure: %w", err)
+	}
+
+	return nil
+}
+
 // fetchAndReset is a convenience method that fetches updates from the remote repository
-// for a specific branch, and then resets the current branch to a specified commit.
-func (r *Repository) fetchAndReset(branch string) error {
-	if err := r.fetch(branch); err != nil {
+// for a specific reference(commit or branch), and then resets the current reference.
+func (r *Repository) fetchAndReset(reference string) error {
+	if err := r.fetch(reference); err != nil {
 		return fmt.Errorf("fetchAndReset failure: %w", err)
 	}
-
-	return r.hardReset(branch)
+	return r.hardReset(reference)
 }
 
 // updateRefSpec updates the reference specification (RefSpec) in the fetch options
 // of the repository operation.
 //   - If a branch name is provided, it sets the RefSpec to fetch that specific branch.
-//   - Otherwise, it sets the RefSpec to fetch all branches.
-//
-// fetching the last commit of one branch is faster than fetching from all branches.
-func (r *Repository) updateRefSpec(branch string) {
+//   - If a commit is provided, it sets the RefSpec to fetch that specific commit.
+//   - If nothing is provided or a invalid commit Hash it will throw an error.
+func (r *Repository) updateRefSpec(reference string) error {
 	var newRefSpec string
 
-	if branch != "" {
-		newRefSpec = fmt.Sprintf("+refs/heads/%s:refs/remotes/origin/%s", branch, branch)
-	} else {
-		newRefSpec = "+refs/heads/*:refs/remotes/origin/*"
+	mode := checkReference(reference)
+	switch mode {
+	case CommitMode:
+		newRefSpec = fmt.Sprintf("+%s:refs/remotes/origin/*", reference)
+	case BranchMode:
+		newRefSpec = fmt.Sprintf("+refs/heads/%s:refs/remotes/origin/%s", reference, reference)
+	case "":
```

Comment:
When will this happen? The checkReference function does not return an empty string

Reply:
Removed